x-pack/plugins/apm/server/routes/profiling/route.ts

```diff
@@ -156,6 +200,148 @@ const profilingFunctionsRoute = createApmServerRoute({
   },
 });
 
+const transactionsFlamegraphRoute = createApmServerRoute({
+  endpoint: 'GET /internal/apm/services/{serviceName}/transactions/flamegraph',
+  params: t.type({
+    path: t.type({ serviceName: t.string }),
+    query: t.intersection([
+      kueryRt,
+      environmentRt,
+      t.type({
+        transactionName: t.string,
+        start: isoToEpochSecsRt,
+        end: isoToEpochSecsRt,
+        transactionType: t.string,
+      }),
+    ]),
+  }),
+  options: { tags: ['access:apm'] },
+  handler: async (resources): Promise<BaseFlameGraph | undefined> => {
+    const { context, plugins, params } = resources;
+    const core = await context.core;
+    const [esClient, profilingDataAccessStart] = await Promise.all([
+      core.elasticsearch.client,
+      await plugins.profilingDataAccess?.start(),
+    ]);
+    if (profilingDataAccessStart) {
+      const { serviceName } = params.path;
+      const {
+        start,
+        end,
+        kuery,
+        transactionName,
+        transactionType,
+        environment,
+      } = params.query;
+
+      return await profilingDataAccessStart?.services.fetchFlamechartData({
+        core,
+        esClient: esClient.asCurrentUser,
+        // TODO: caue fix this
+        indices: '.ds-traces-apm-default-2024.02.02-000001',
```

Comment:
Q: Like the TODO :D Which index should be the correct one here? I tried to test it with a remote cluster and got 500 index_not_found_exception but it could be something in my setup.

Reply:
Thanks, missed that.